### ISSUE TITLE
doublespend attempt error communicate through API

### ISSUE THREAD

**Issue by carryforward** [btcsuite/btcwallet#334]
There are tons of reorgs happening on testnet.  it happened to break our software.  it would be nice to have this error propagated up through the API.  

I ran the btcctl sendtoaddress command and got this error:

```
-32603: -22: TX rejected: output dfcee8223b3cde3095133c710ec44629dbcc29d30da42befaa767d190da54abe:1 already spent by transaction cb299047c11086fc6fa571d7cf92d03864a17463a5282edd56d5d0d13ed902a7 in the memory pool
```

It would have saved a bunch of time if it were also passed as an error on the API.

**Comment by jrick** [comment]
This may be futile since blocks (and possibly double spends) are always being added but can you post output of these commands?

```
btcctl --testnet --wallet gettransaction cb299047c11086fc6fa571d7cf92d03864a17463a5282edd56d5d0d13ed902a7 
btcctl --testnet --wallet gettransaction dfcee8223b3cde3095133c710ec44629dbcc29d30da42befaa767d190da54abe
btcctl --testnet --wallet listunspent | less # just search for dfcee8223b3cde3095133c710ec44629dbcc29d30da42befaa767d190da54abe with vout 1 and see if it's listed as a utxo or not
```

Also could you describe what your application is using wallet for?  It would help to know what wallet was doing during this time and how it broke your app.

**Comment by carryforward** [comment]
shoot, I restarted btcwallet, but I think I know what is happening.  cb299 is gone now

We have 28 or so UTXOs in the wallet.  we spent all of them and they got included in a block.  Testnet then reorged and our transactions were not in the new branch.  They are still echoing around the network though.  It has been several hours, and they haven't gotten back to a miner nor to the block explorers I am checking.

here are a couple of other different orphaned transactions which also haven't been put back in blocks.

https://www.blocktrail.com/tBTC/tx/5674fd4d5bdd4299f06c163938b1d0d61b9c905deb444b60843fbb10fdef2ce3

https://www.blocktrail.com/tBTC/tx/798bee43c21d9fe8c9b4f486a0060c4ffe5e6e55be79d9f63e41478393d9d9b1

```
btcctl --testnet  --wallet gettransaction 5674fd4d5bdd4299f06c163938b1d0d61b9c905deb444b60843fbb10fdef2ce3
-5: No information for transaction
```

```
btcctl --testnet --wallet gettransaction dfcee8223b3cde3095133c710ec44629dbcc29d30da42befaa767d190da54abe
{
  "amount": 0.06266,
  "fee": 0.12533,
  "confirmations": 0,
  "blockhash": "",
  "blockindex": 0,
  "blocktime": 0,
  "txid": "dfcee8223b3cde3095133c710ec44629dbcc29d30da42befaa767d190da54abe",
  "walletconflicts": [],
  "time": 1448520040,
  "timereceived": 1448520040,
  "details": [
    {
      "account": "",
      "amount": -0.06267,
      "category": "send",
      "fee": 0.12533,
      "vout": 0
    },
    {
      "account": "",
      "address": "mrM9rnFUTZfsxgw6awLr2kRYnLmGHTQ5H2",
      "amount": 0.06266,
      "category": "receive",
      "vout": 1
    }
  ],
  "hex": "01000000010215b2207182427bec291d18dd316af0b088c08b060091bab8b7525084207903010000006b483045022100d2c1d1a0f506508422e7d160b083b6c61b122f57657fc6108484cc891721fa7e0220427338f72bce677afae50f1e41a2b2c8f1fbaf0db66bab3ec670106e6a93e80a012102cd3f2f1ab9e89518036e1597e5b59777fbb16df553f6437e5ab40429f497b489ffffffff0200000000000000002a6a284661000000000ca49117e3582d62e97b3f484eea849037feac94ebda17a0eaf0e3cad98e10c98006909c5f00000000001976a91476cee09aaaf961deff220e74a032c398cae5526c88ac00000000"
}

```

hmm, thats odd, it looks like it is paying a fee larger than the inputs.  probably just a display thing.

outpoint 0 should be an op_return with a zero value output.

6a28 <opreturn instructions
4661000000000ca417e3582d62e97b3f484eea849037feac94ebda17a0eaf0e3cad98e10c9800690 <opreturn payload


BTW, I am running btcsuitereleases , which is a few months behind.  .12 is now ready, so maybe its time to update soon.
https://github.com/btcsuitereleases/btcwallet





grr, while I was writing this my btcd jumped around due to the time bip voting.
```
07:23:33 2015-11-26 [INF] BMGR: Processed 1 block in the last 1m25.93s (1 transaction, height 607397, 2015-11-26 07:22:14 +0000 UTC)
07:27:42 2015-11-26 [INF] CHAN: REORGANIZE: Block 0000000000000441980e01b11f281f15b2d01097ca590f1eb20a60958638bf99 is causing a reorganize.
07:28:29 2015-11-26 [INF] CHAN: REORGANIZE: Chain forks at 0000000000000248d8f59e8761223ead9d2f163a72e562dc223d4f2f53db2071
07:28:29 2015-11-26 [INF] CHAN: REORGANIZE: Old best chain head was 000000003aae2dba1c5e83b83ac4e554ee330cd04083b8340066cd091f2619b5
07:28:29 2015-11-26 [INF] CHAN: REORGANIZE: New best chain head is 0000000000000441980e01b11f281f15b2d01097ca590f1eb20a60958638bf99
07:28:29 2015-11-26 [INF] BMGR: Processed 1 block in the last 4m56.25s (1804 transactions, height 607398, 2015-11-26 07:21:59 +0000 UTC)
07:43:30 2015-11-26 [INF] BMGR: Processed 1 block in the last 15m0.22s (1779 transactions, height 607399, 2015-11-26 07:42:10 +0000 UTC)
07:44:17 2015-11-26 [INF] CHAN: FORK: Block 000000003b6447d8cd6ebdedd7bb2f60ab350f82e75d87066195d603bcbcf042 forks the chain at height 607398/block 0000000000000441980e01b11f281f15b2d01097ca590f1eb20a60958638bf99, but does not cause a reorganize
07:44:17 2015-11-26 [INF] BMGR: Processed 1 block in the last 47.95s (1 transaction, height 607399, 2015-11-26 07:42:33 +0000 UTC)
```
Now the dfcee transaction isn't even in the blockchain anymore.

What a mess.

basically, I have a bunch of transactions which are floating around in mempools and are included (or not) in the different branches.  I guess my main problem is that they are not being confirmed.  Long story short, my colleague was printing out returned errors and he says there was nothing to print which would have helped him diagnose this without resorting to btcctl.

Here is the code we are running:
https://github.com/FactomProject/FactomCode/blob/anchor_btcd/anchor/anchor.go#L104
https://github.com/FactomProject/FactomCode/blob/anchor_btcd/anchor/anchor.go#L280

**Comment by jrick** [comment]
Both of those transactions have been mined now and have > 150 confirmations.  There's little a wallet can do if miners are not including a transaction, except wait.  sendrawtransaction can't error since the send was successful (the transaction was accepted to your btcd mempool and btcd inv'd it to its peers).

Since there are no btcwallet notifications for when a transaction is mined, orphaned, or double spent you will have to query the wallet for that information.  I am working on a new RPC API and it includes notifications for when transactions are mined and reorged on a chain switch, but the old RPC server requires ugly workarounds to use it correctly.

**Comment by carryforward** [comment]
eh, maybe I have too much context.

The request was to have the error `TX rejected: output dfcee8223b3cde3095133c710ec44629dbcc29d30da42befaa767d190da54abe:1 already spent by transaction cb299047c11086fc6fa571d7cf92d03864a17463a5282edd56d5d0d13ed902a7 in the memory pool`  be returned with the error message.

It is unclear where the bug is at this point, and it did take me reimporting the bip32 seed to make the wallet forget the transactions.  This error probably wont even be seen in normal cases.

I will close this issue until I have a better way to exercise the bug.

**Comment by davecgh** [comment]
For future reference, there is a `dropwtxmgr` tool in the `cmd` directory which will let you drop all transactions without having to redo an import.
